test/if.spec.js

```diff
@@ -134,13 +119,13 @@ describe('if', () => {
       sut.view = new ViewMock();  
       sut.view.isBound = true;        
       sut.viewSlot.children = [];
-      sut.valueChanged(true);        
+      sut.conditionChanged(true);        
    
       delay(200).then(() => {          
-        sut.valueChanged(false);
+        sut.conditionChanged(false);
         return delay(400);
       }).then(() => {
-        sut.valueChanged(true);
+        sut.conditionChanged(true);
         return delay(600);          
       }).then(() => {  
         expect(viewSlot.children.length).toEqual(1);
```

Comment:
@jods4 This assertion is failing for me. Does it work for you?

Reply:
@jdanyow I did not look specifically but when I committed all `if` tests were (seemed?) green :(

I am away from code but if I should make a guess: I wonder if the problem is here:
https://github.com/aurelia/templating-resources/blob/master/src/if.js#L127

The thing that is broken in tests is that they call `conditionChanged` (`valueChanged` before) but they do not actually set `condition`, which can lead to discrepencies.

`conditionChanged` is an internal callback when `condition` changes, it's not meant to be used from outside.
What looks right to me would be `condition = false`  instead of `conditionChanged(false)`. For some reason I didn't have time to debug, this doesn't work at all even when flushing the micro-task queue just after. Not sure why, I thought it would work.